curve/ecgfp5/scalar_field.go

```diff
@@ -0,0 +1,443 @@
+package ecgfp5
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math/big"
+	"math/bits"
+	"math/rand"
+	"time"
+
+	gFp5 "github.com/elliottech/poseidon_crypto/field/goldilocks_quintic_extension"
+)
+
+// ECgFp5Scalar represents the scalar field of the ECgFP5 elliptic curve where
+// p = 1067993516717146951041484916571792702745057740581727230159139685185762082554198619328292418486241
+type ECgFp5Scalar struct {
+	Value [5]big.Int
```

Comment:
why is this big int? I think these are goldilocks elements, so uint64 should be ok